functions.sh

```diff
@@ -128,7 +128,18 @@ function do_dump() {
       DB_NAMES=$(mysql -h $DB_SERVER -P $DB_PORT $DBUSER $DBPASS -N -e 'show databases')
       [ $? -ne 0 ] && return 1
     fi
+    if [[ -n "$DB_NAMES_EXCLUDE" ]]; then
+      # exclude databases if specificied
+      DB_NAMES_EXCLUDE=(`echo $DB_NAMES_EXCLUDE`)
```

Comment:
Can we use a new variable name. I tend to dislike reusing a variable for a different structure and purpose. It makes debugging hard, some linters do not allow it.

Reply:
You shouldn't need the `echo` here; the following should work fine:

```bash
NEWVAR=($DB_NAMES_EXCLUDE)
```

Reply:
I think this would be easier with an associative array:

```bash
if [ -z "$DB_NAMES_EXCLUDE" ]; then
   DB_NAMES_EXCLUDE="information_schema performance_schema mysql sys"
fi
declare -A exclude_list
for i in $DB_NAMES_EXCLUDE; do
  exclude_list[$i]="true"
done
for onedb in $DB_NAMES; do
  if [ -n ${exclude_list[$onedeb]} ]; then
    continue
  fi
```

That might need some tweaking, but should do it.

Reply:
I've did it the way you suggested but instead of "-n" I tested with "-v" because "-n" causes issues in the test-statement if the exclude_list[$onedb] is not defined.

Reply:
Sure, that is fine